### PR TITLE
Could you please check the tests and give me feedback.

### PR DESCRIPTION
**Pull request by invalidred**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -41,7 +41,10 @@
     "gulp": "3.9.0",
     "gulp-babel": "5.1.0",
     "gulp-mocha": "2.1.2",
-    "mocha": "2.2.5"
+    "mocha": "^2.2.5",
```
Comment (anvk) [inline package.json]: I was thinking just to follow the suite and only keep strict versions without ^

File: src/easy-sql-test.js [hunk 2]
```diff
@@ -39,6 +43,10 @@ export default class EasySQLTest {
   }
 
   _executeStorProc(storProcName, args = {}, callback) {
+    if (!storProcName) {
```
Comment (anvk) [inline src/easy-sql-test.js:46]: we do this check for `_executeStorProc()` but not for `_query()` function. I think that we should do for both or for none.

File: src/easy-sql-test.js [hunk 3]
```diff
@@ -63,6 +70,10 @@ export default class EasySQLTest {
     let result = [];
 
     for (let query of prepQueries) {
+      if (!query || /^\s+$/.test(query)) {
```
Comment (anvk) [inline src/easy-sql-test.js]: I think we should just form those objects "as is". Then if one of the queries will fail inside of `_query()` function well maybe then developer will see throw. I just thinking of a very complex case when array of queries is being generated dynamically. It would be a very subtle bug if one of the generated queries is empty but then it would be "skipped" by the `_convertQueriesToTestSteps()`. Also maybe it makes sense to rename argument to `queries` from `prepQueries`

File: test/easy-sql-test.js [hunk 4]
```diff
@@ -2,13 +2,742 @@
 
 var chai = require('chai'),
     expect = chai.expect,
-    googleStocks = require('../dist/easy-sql-test.js');
+    EasySqlTest = require('../dist/easy-sql-test.js'),
+    proxyquire = require('proxyquire'),
+    sinon = require('sinon');
 
 describe('easy-sql-test tests', function() {
+  var emptyDbConfig = {};
```
Comment (anvk) [inline test/easy-sql-test.js]: I would rename it to `someDbConfig`.

File: test/easy-sql-test.js [hunk 5]
```diff
@@ -2,13 +2,742 @@
...
+      easySqlTestProxy.connectionOpen(connectCallback);
+
+      expect(connectCallback.called).to.be.true;
+      expect(connectCallback.callCount).to.equal(1);
+    });
+  });
+
+  describe('connectionClose tests', function(){
+    it('check connection.close() called', function() {
+      var closeFunctionSpy = sinon.spy(function(){});
```
Comment (anvk) [inline test/easy-sql-test.js]: `var closeFunctionSpy = sinon.spy(function() {});`

File: test/easy-sql-test.js [hunk 6]
```diff
@@ -2,13 +2,742 @@
...
+      var easySqlTestProxy = new EasySqlTest(emptyDbConfig);
+
+      easySqlTestProxy.connectionOpen(connectCallback);
+
+      expect(connectCallback.called).to.be.true;
+      expect(connectCallback.callCount).to.equal(1);
+    });
+  });
+
+  describe('connectionClose tests', function(){
```
Comment (anvk) [inline test/easy-sql-test.js]: `describe('connectionClose tests', function() {`

File: test/easy-sql-test.js [hunk 7]
```diff
@@ -2,13 +2,742 @@
...
+    it('check connection.close() called', function() {
+      var closeFunctionSpy = sinon.spy(function(){});
+      var mssqlStub = {
+        Connection: function() {
+          return {
+            connect: function(){},
```
Comment (anvk) [inline test/easy-sql-test.js]: `connect: function() {},`

File: test/easy-sql-test.js [hunk 8]
```diff
@@ -2,13 +2,742 @@
...
+
+      easySqlTestProxy.connectionOpen();
+      var actualConnection = easySqlTestProxy.connection;
+      expect(actualConnection).to.deep.equal(expectedConnection);
+    });
+
+    it('get dbConfig()', function() {
+
+    	var expected = {
+    		a: 'a',
```
Comment (anvk) [inline test/easy-sql-test.js]: extra space ?

File: test/easy-sql-test.js [hunk 9]
```diff
@@ -2,13 +2,742 @@
...
+    	};
+
+    	var easySqlTest = new EasySqlTest(expected);
+
+    	var actual = easySqlTest.dbConfig;
+    	expect(expected).to.deep.equal(actual);
+    });
+  });
+
+  describe('_executeStorProc() tests', function(){
```
Comment (anvk) [inline test/easy-sql-test.js]: `describe('_executeStorProc() tests', function() {`

File: test/easy-sql-test.js [hunk 10]
```diff
@@ -2,13 +2,742 @@
...
+
+    it('no storProcName should throw error', function() {
+  		var errorMessage = 'easy-sql-test: _executeStorProc() requires' +
```
Comment (anvk) [inline test/easy-sql-test.js]: I think your editor is messed up. Could you please double check not to use Tabs anywhere in the file please

File: test/easy-sql-test.js [hunk 11]
```diff
@@ -2,13 +2,742 @@
...
+
+      try{
+  			easySqlTest._executeStorProc();
+  		} catch(err) {
+  			expect(err).to.equal(errorMessage);
+  		}
+  	});
+
+    it('connection.request.input() should be called once per arg', function() {
+        var inputFunctionSpy = sinon.spy(function() {});
```
Comment (anvk) [inline test/easy-sql-test.js:129]: could you also please ensure to use 2 space tabs and not 4 space

File: test/easy-sql-test.js [hunk 12]
```diff
@@ -2,13 +2,742 @@
...
+        this._errorCallback(error);
+      }
+    };
+
+    beforeEach(function() {
+      easySqlTest = new EasySqlTest(emptyDbConfig);
+      prepQuery = undefined;
+    });
+
+    it('with no prepQuery', function() {
```
Comment (anvk) [inline test/easy-sql-test.js]: if we change prepQuery to queries then we need to change name here. Also tests should reflect decisions I mentioned ^ in case if we want to avoid skipping queries which are empty or "malformed"

File: package.json (hunk 1 above)
Comment (invalidred) [inline package.json]: updated code

File: src/easy-sql-test.js (hunk 2 above)
Comment (invalidred) [inline src/easy-sql-test.js:46]: good point, added check for _query() function and updated unit tests accordingly

File: src/easy-sql-test.js (hunk 3 above)
Comment (invalidred) [inline src/easy-sql-test.js]: removed white space regex and renamed args to queries.

File: test/easy-sql-test.js (hunk 4 above)
Comment (invalidred) [inline test/easy-sql-test.js]: renamed emptyDbConfig to someDbConfig

File: test/easy-sql-test.js (hunk 5 above)
Comment (invalidred) [inline test/easy-sql-test.js]: fixed spacing

File: test/easy-sql-test.js (hunk 6 above)
Comment (invalidred) [inline test/easy-sql-test.js]: fixed spacing

File: test/easy-sql-test.js (hunk 7 above)
Comment (invalidred) [inline test/easy-sql-test.js]: fixed spacing

File: test/easy-sql-test.js (hunk 8 above)
Comment (invalidred) [inline test/easy-sql-test.js]: fixed spacing

File: test/easy-sql-test.js (hunk 9 above)
Comment (invalidred) [inline test/easy-sql-test.js]: fixed spacing

File: test/easy-sql-test.js (hunk 10 above)
Comment (invalidred) [inline test/easy-sql-test.js]: converted tabs to spaces

File: test/easy-sql-test.js (hunk 11 above)
Comment (invalidred) [inline test/easy-sql-test.js:129]: updated.

File: test/easy-sql-test.js (hunk 12 above)
Comment (invalidred) [inline test/easy-sql-test.js]: renamed prepQuery to queires. I did remove the empty space regex from the comment above. However there is still a (!query) { continue } instruction. Should this be removed as well?

File: src/easy-sql-test.js [hunk 13]
```diff
@@ -50,19 +58,26 @@ export default class EasySQLTest {
...
     return request.query(query, callback);
   }
 
-  _convertQueriesToTestSteps(prepQueries = []) {
+  _convertQueriesToTestSteps(queries = []) {
     let result = [];
 
-    for (let query of prepQueries) {
+    for (let query of queries) {
+      if (!query) {
```
Comment (anvk) [inline src/easy-sql-test.js]: I think we should remove this block. Let `_query()` function to fail and notify developer about it instead of removing a step from the array. Silent removal is way worse then loud fail here

Comment (invalidred) [inline src/easy-sql-test.js]: Makes sense. Updated code and unit tests accordingly.

File: .travis.yml [hunk 14]
```diff
@@ -1,3 +1,3 @@
 language: node_js
 node_js:
-  - "0.10"
\ No newline at end of file
+  - "0.12"
```
Comment (anvk) [inline .travis.yml:5]: I specified in the `package.json` that lib works with Node 10 and higher. I would assume that tests should be ran on 0.10, 0.11 and 0.12 node versions

http://docs.travis-ci.com/user/languages/javascript-with-nodejs/

Comment (invalidred) [inline .travis.yml:5]: The issue was for-of loop that used Symbols which doesn't work with Node 10.XX. and 11.XX. Updated code to use forEach loop instead of for-of loop to solve the issue.

**Comment by invalidred** [comment]
Updated code as per code review.

**Comment by invalidred** [comment]
Added version node .10, .11 and .12 to travis and the tests pass
https://travis-ci.org/anvk/easy-sql-tests/builds/74079390

**Comment by invalidred** [comment]
Added comment to use for-of loop when there is better support for it in Babel and Node.js
